From 69e371883ad60893be7f691c74e8f0a66b2ea87d Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Thu, 19 Dec 2024 11:19:50 +0800
Subject: [PATCH 151/210] mtk: hostapd: rename mld_id to requested_mld_ap in
 probe_resp_params for readability

The renamed 'requested_mld_ap' is only set when handle an ML probe
request, and is used in the following handling for ML probe response
generation.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/beacon.c | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/src/ap/beacon.c b/src/ap/beacon.c
index 4dca43e59..fcb33e8f4 100644
--- a/src/ap/beacon.c
+++ b/src/ap/beacon.c
@@ -714,7 +714,7 @@ struct probe_resp_params {
 	bool is_p2p;
 
 	/* Generated IEs will be included inside an ML element */
-	struct hostapd_data *mld_ap;
+	struct hostapd_data *requested_mld_ap;
 	struct mld_info *mld_info;
 
 	struct ieee80211_mgmt *resp;
@@ -775,7 +775,7 @@ static size_t hostapd_probe_resp_elems_len(struct hostapd_data *hapd,
 #ifdef CONFIG_IEEE80211BE
 	if (hapd->iconf->ieee80211be && !hapd->conf->disable_11be) {
 		struct hostapd_data *ml_elem_ap =
-			params->mld_ap ? params->mld_ap : hapd;
+			params->requested_mld_ap ? params->requested_mld_ap : hapd;
 
 		buflen += hostapd_eid_eht_capab_len(hapd, IEEE80211_MODE_AP);
 		buflen += 3 + sizeof(struct ieee80211_eht_operation);
@@ -784,7 +784,8 @@ static size_t hostapd_probe_resp_elems_len(struct hostapd_data *hapd,
 
 		if (ml_elem_ap->conf->mld_ap) {
 			buflen += hostapd_eid_eht_ml_beacon_len(
-				ml_elem_ap, params->mld_info, !!params->mld_ap);
+				ml_elem_ap, params->mld_info,
+				!!params->requested_mld_ap);
 
 			/* For Max Channel Switch Time element during channel
 			 * switch */
@@ -938,12 +939,12 @@ static u8 * hostapd_probe_resp_fill_elems(struct hostapd_data *hapd,
 #ifdef CONFIG_IEEE80211BE
 	if (hapd->iconf->ieee80211be && !hapd->conf->disable_11be) {
 		struct hostapd_data *ml_elem_ap =
-			params->mld_ap ? params->mld_ap : hapd;
+			params->requested_mld_ap ? params->requested_mld_ap : hapd;
 
 		if (ml_elem_ap->conf->mld_ap)
 			pos = hostapd_eid_eht_ml_beacon(
 				ml_elem_ap, params->mld_info,
-				pos, !!params->mld_ap);
+				pos, !!params->requested_mld_ap);
 
 		pos = hostapd_eid_eht_capab(hapd, pos, IEEE80211_MODE_AP);
 		pos = hostapd_eid_eht_operation(hapd, pos);
@@ -1072,7 +1073,7 @@ static void hostapd_fill_probe_resp_ml_params(struct hostapd_data *hapd,
 {
 	struct hostapd_data *link;
 
-	params->mld_ap = NULL;
+	params->requested_mld_ap = NULL;
 	params->mld_info = os_zalloc(sizeof(*params->mld_info));
 	if (!params->mld_info)
 		return;
@@ -1092,7 +1093,7 @@ static void hostapd_fill_probe_resp_ml_params(struct hostapd_data *hapd,
 		 * interface.
 		 */
 		if (mld_id != -1 && link->iface == hapd->iface)
-			params->mld_ap = link;
+			params->requested_mld_ap = link;
 
 		/* Never duplicate main Probe Response frame body */
 		if (link == hapd)
@@ -1111,7 +1112,7 @@ static void hostapd_fill_probe_resp_ml_params(struct hostapd_data *hapd,
 			   mld_link_id, link_info->resp_sta_profile_len);
 	}
 
-	if (mld_id != -1 && !params->mld_ap) {
+	if (mld_id != -1 && !params->requested_mld_ap) {
 		wpa_printf(MSG_DEBUG,
 			   "MLD: No nontransmitted BSSID for MLD ID %d",
 			   mld_id);
@@ -1122,7 +1123,7 @@ static void hostapd_fill_probe_resp_ml_params(struct hostapd_data *hapd,
 
 fail:
 	hostapd_free_probe_resp_params(params);
-	params->mld_ap = NULL;
+	params->requested_mld_ap = NULL;
 	params->mld_info = NULL;
 }
 #endif /* CONFIG_IEEE80211BE */
@@ -1780,7 +1781,7 @@ static u8 * hostapd_probe_resp_offloads(struct hostapd_data *hapd,
 	params.is_p2p = false;
 	params.known_bss = NULL;
 	params.known_bss_len = 0;
-	params.mld_ap = NULL;
+	params.requested_mld_ap = NULL;
 	params.mld_info = NULL;
 
 	hostapd_gen_probe_resp(hapd, &params);
@@ -1823,7 +1824,7 @@ u8 * hostapd_unsol_bcast_probe_resp(struct hostapd_data *hapd,
 	probe_params.is_p2p = false;
 	probe_params.known_bss = NULL;
 	probe_params.known_bss_len = 0;
-	probe_params.mld_ap = NULL;
+	probe_params.requested_mld_ap = NULL;
 	probe_params.mld_info = NULL;
 
 	hostapd_gen_probe_resp(hapd, &probe_params);
-- 
2.45.2

